tests/integration/solidity_contracts/UniswapV2/test_uniswap_v2_factory.py

```diff
@@ -80,39 +68,39 @@ async def test_should_create_pair_only_once(
                 }
             ]
 
-            with pytest.raises(Exception) as e:
+            with kakarot_error("UniswapV2: PAIR_EXISTS"):
                 await factory.createPair(
                     *tokens,
                     caller_address=owner.starknet_address,
                 )
-            message = re.search(r"Error message: (.*)", e.value.message)[1]  # type: ignore
-            # TODO: update with https://github.com/sayajin-labs/kakarot/issues/416
-            assert message == "Kakarot: Reverted with reason: 147028384"
 
-            with pytest.raises(Exception) as e:
+            with kakarot_error("UniswapV2: PAIR_EXISTS"):
                 await factory.createPair(
                     *tokens[::-1],
                     caller_address=owner.starknet_address,
                 )
-            message = re.search(r"Error message: (.*)", e.value.message)[1]  # type: ignore
-            # TODO: update with https://github.com/sayajin-labs/kakarot/issues/416
-            assert message == "Kakarot: Reverted with reason: 147028384"
 
-            assert await factory.getPair(*tokens) == create2_address
-            assert await factory.getPair(*tokens[::-1]) == create2_address
-            assert await factory.allPairs(0) == create2_address
+            assert await factory.getPair(*tokens) == pair_evm_address
+            assert await factory.getPair(*tokens[::-1]) == pair_evm_address
+            assert await factory.allPairs(0) == pair_evm_address
             assert await factory.allPairsLength() == 1
 
+            salt = keccak(encode_abi(["address", "address"], sorted(tokens)))
+            pair_starknet_address = await get_starknet_address(salt)
+            pair = get_solidity_contract(
+                "UniswapV2", "UniswapV2Pair", pair_starknet_address, pair_evm_address
+            )
             assert await pair.factory() == factory.evm_contract_address
             assert await pair.token0() == tokens[0]
             assert await pair.token1() == tokens[1]
 
+        @pytest.mark.skip("gas_usage is not yet returned by kakarot")
         async def test_should_use_correct_gas(self, factory, owner):
+            # TODO: see https://github.com/sayajin-labs/kakarot/issues/428
             tx = await factory.createPair(
                 *TEST_ADDRESSES, caller_address=owner.starknet_address
             )
-            # TODO: see https://github.com/sayajin-labs/kakarot/issues/428
-            # assert tx.gasUsed == 2512920
+            assert tx.gas_used == 2512920
```

Comment:
Should we assert here with a hardcoded number? If somehow we alter Kakarot's gas logic in the coming weeks, this test will start failing

Reply:
this is however the univ2 test as is

Reply:
see https://github.com/Uniswap/v2-core/blob/ee547b17853e71ed4e0101ccfd52e70d5acded58/test/UniswapV2Factory.spec.ts#L71